stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -875,89 +389,58 @@ public class MessageListViewModel(
         val reaction = Reaction().apply {
             messageId = message.id
             type = reactionType
             score = 1
         }
-        if (message.ownReactions.any { it.type == reactionType }) {
-            chatClient.deleteReaction(
-                messageId = message.id,
-                reactionType = reaction.type,
-                cid = cid
-            ).enqueue(
-                onError = { chatError ->
-                    logger.e {
-                        "Could not delete reaction for message with id: ${reaction.messageId} " +
-                            "Error: ${chatError.message}. Cause: ${chatError.cause?.message}"
-                    }
-                }
-            )
-        } else {
-            chatClient.sendReaction(
-                enforceUnique = enforceUnique,
-                reaction = reaction,
-                cid = cid
-            ).enqueue(
-                onError = { chatError ->
-                    logger.e {
-                        "Could not send reaction for message with id: ${reaction.messageId} " +
-                            "Error: ${chatError.message}. Cause: ${chatError.cause?.message}"
-                    }
-                }
-            )
-        }
+        messageListController.reactToMessage(reaction, message)
     }
 
     /**
-     * Called when upon initialization or exiting thread mode.
+     * Called when upon initialization or when exiting thread mode.
      */
     private fun onNormalModeEntered() {
-        currentMode = Mode.Normal
-        resetThread()
+        messageListController.enterNormalMode()
     }
 
     /**
      * Sets the value used to filter deleted messages.
      * @see DeletedMessageVisibility
      *
-     * @param deletedMessageVisibility Changes the visibility of deleted messages.
+     * @param deletedMessageVisibility Determines the visibility of deleted messages.
      */
     public fun setDeletedMessageVisibility(deletedMessageVisibility: DeletedMessageVisibility) {
-        this._deletedMessageVisibility.value = deletedMessageVisibility
+        messageListController.setDeletedMessageVisibility(deletedMessageVisibility)
     }
 
     /**
      * Sets the value used to determine if message footer content is shown.
      * @see MessageFooterVisibility
      *
-     * @param messageFooterVisibility Changes the visibility of message footers.
+     * @param messageFooterVisibility Determines the visibility of message footers.
      */
     public fun setMessageFooterVisibility(messageFooterVisibility: MessageFooterVisibility) {
-        this.messageFooterVisibility.value = messageFooterVisibility
+        messageListController.setMessageFooterVisibility(messageFooterVisibility)
+    }
+
+    /**
+     * Sets whether the system messages should be visible.
+     *
+     * @param showSystemMessages Whether system messages should be visible or not.
+     */
+    public fun setAreSystemMessagesVisible(showSystemMessages: Boolean) {
```

Comment:
If we adopt the naming scheme suggested in the controller comment, it should be reflected here as well.